packages/core/saved-objects/core-saved-objects-migration-server-internal/src/core/build_active_mappings.test.ts

```diff
@@ -58,19 +63,32 @@ describe('buildActiveMappings', () => {
     expect(buildActiveMappings(typeMappings)).toMatchSnapshot();
   });
 
-  test('generated hashes are stable', () => {
+  test(`includes the provided override properties, except for 'properties'`, () => {
     const properties = {
       aaa: { type: 'keyword', fields: { a: { type: 'keyword' }, b: { type: 'text' } } },
       bbb: { fields: { b: { type: 'text' }, a: { type: 'keyword' } }, type: 'keyword' },
       ccc: { fields: { b: { type: 'text' }, a: { type: 'text' } }, type: 'keyword' },
     } as const;
 
-    const mappings = buildActiveMappings(properties);
-    const hashes = mappings._meta!.migrationMappingPropertyHashes!;
+    const someOverrideMappings: Partial<IndexMapping> = {
+      dynamic: true, // just to illustrate override works, not something we want to do
+      _meta: {
+        migrationMappingPropertyHashes: {
+          foo: 'someLongHash',
+          bar: 'anotherLongHash',
+          baz: '10.3.0', // hashes and versions will NOT coexist (it's either one or the other)
```

Comment:
Thinking about it, I think I'll switch to the same property names used on ZDT algorithm: `appVersions` and `docVersions`. This way, I will not impact the existing hashes field (which will simply be deprecated).

I believe the advantages of doing this are worth it:
* We will not be reusing a property with an outdated / misleading name.
* We will align with the _meta stored by the ZDT algorithm, which could, in the future, facilitate transition from Cloud to Serverless.
* If a migration fails or Kibana fails to start after a migration, and a rollback is performed, the previous version using hashes will still have the hashes field in place. This will prevent the migration logic from picking up ALL types (this could be a costly operation for large customers).